File: src/app/models/document.model.ts
import { PageModel } from './page.model';

export type UUID = string;

export interface DocumentModel {
  id: UUID;
  title: string;
  version: string;
  pageSize: PageSize;
  sections: SectionModel[];
  metadata?: Record<string, unknown>;
  header?: HeaderConfig;
  footer?: FooterConfig;
  logo?: LogoConfig;
}

export interface HeaderConfig {
  leftText?: string;
  centerText?: string;
  rightText?: string;
  leftImage?: string; // base64 or URL
  centerImage?: string;
  rightImage?: string;
  textColor?: string;
  showPageNumber?: boolean;
  pageNumberFormat?: 'arabic' | 'roman' | 'alphabetic'; // 1,2,3 | i,ii,iii | a,b,c
}

export interface FooterConfig {
  leftText?: string;
  centerText?: string;
  centerSubText?: string;
  leftImage?: string; // base64 or URL
  centerImage?: string;
  rightImage?: string;
  textColor?: string;
  showPageNumber?: boolean;
  pageNumberFormat?: 'arabic' | 'roman' | 'alphabetic'; // 1,2,3 | i,ii,iii | a,b,c
}

export interface LogoConfig {
  url?: string;
  position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
}

export interface PageSize {
  widthMm: number;
  heightMm: number;
  dpi?: number;
}

export interface SectionModel {
  id: UUID;
  title: string;
  subsections: SubsectionModel[];
}

export interface SubsectionModel {
  id: UUID;
  title: string;
  pages: PageModel[];
}

export interface HierarchySelection {
  sectionId: string | null;
  subsectionId: string | null;
  pageId: string | null;
}

export interface SubsectionSelection {
  subsectionId: string | null;
  pageId: string | null;
}

File: report-generator-backend/src/main/java/com/org/report_generator/model/document/DocumentModel.java
package com.org.report_generator.model.document;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class DocumentModel {
    private String id;
    private String title;
    private String version;
    private String schemaVersion;
    private PageSize pageSize = new PageSize();
    private List<Section> sections = new ArrayList<>();
    private Map<String, Object> metadata;
    private HeaderConfig header;
    private FooterConfig footer;
    private LogoConfig logo;
}



